### PR TITLE
Add object fusion tracks, fix sidebar names

### PR DESCRIPTION
**Pull request by nmhaddadND**
This PR adds support for Object Fusion Tracks and renames some items in our sidebar so that everything is correct.

### CODE REVIEW

File: demo/objectFusionTracksLoader.js [hunk 1]
```diff
@@ -1,215 +1,386 @@
...
+          } else {
+            const FlatbufferModule = await import(schemaUrl);
+            const trackGeometries = await parseTracks(data.Body, shaderMaterial, FlatbufferModule, animationEngine);
+            await callback(trackGeometries, );
+          }
+          incrementLoadingBarTotal("tracks loaded")
+        });
+      request.on("httpDownloadProgress", async (e) => {
+        await updateLoadingBar(e.loaded/e.total * 100);
+      });
```
Comment (bcasselsND) [inline demo/objectFusionTracksLoader.js:52]: You don't need `async` or `await` here. The progress thing gets called, and no one is waiting for the answer.

File: demo/objectFusionTracksLoader.js [hunk 2]
```diff
@@ -1,215 +1,386 @@
...
-      // Initializations:
-      const centroidLocation = new THREE.Vector3( detect.centroid().x(), detect.centroid().y(), detect.centroid().z() );
-      if (firstCentroid == undefined) {
-        firstCentroid = centroidLocation;
-      }
+      function getBoundingBoxGeometry(t0, state, material) {
 
-      delta = centroidLocation.clone().sub(firstCentroid);
+        // Initializations:
+        let centroidLocation;
```
Comment (bcasselsND) [inline demo/objectFusionTracksLoader.js:185]: Why is this `let` here? Why not just `const` below, where you get the value?

**Review by bcasselsND — CHANGES_REQUESTED** [review]
`let` to `const`, wherever yo ucan.

**Comment by nmhaddadND** [comment]
I will make these changes, but just to be clear, this is more than likely a temporary fix for getting object fusion bounding boxes displayed in potree under the correct sidebar subtree. I copied and pasted almost all code from `trackLoader.js`, which @jallen-NextDroid is in the middle of redoing.

File: demo/objectFusionTracksLoader.js [hunk 3]
```diff
@@ -1,215 +1,304 @@
...
-        const bufferBoxGeometry = new THREE.BufferGeometry().fromGeometry(allBoxes);
-        const edges = new THREE.EdgesGeometry( bufferBoxGeometry ); // or WireframeGeometry( geometry )
-        const timestamps = [];
-        for (let tt=0, numTimes=detectTimes.length; tt<numTimes; tt++) {
-          for (let kk=0, numVerticesPerBox=24; kk<numVerticesPerBox; kk++) {  // NOTE: 24 vertices per edgesBox
-            timestamps.push(detectTimes[tt]);
+        if (t0 == -1) {
+          t0 = state.timestamps();
+        }
+        stateTimes.push(state.timestamps()-animationEngine.tstart); // HACK -- 16.8 is a hack to get the tracked box timestamps to lineup with the rest of the animation
```
Comment (bcasselsND) [inline demo/objectFusionTracksLoader.js:197]: Is the comment still relevant? I see no 16.8 in the code.

File: demo/objectFusionTracksLoader.js [hunk 4]
```diff
@@ -1,215 +1,304 @@
...
+	await loadTracks(s3, bucket, name, trackFileName, trackShaderMaterial, animationEngine, (trackGeometries) => {
+		const objectFusionTrackLayer = new THREE.Group();
+    objectFusionTrackLayer.name = trackName;
+    objectFusionTrackLayer.visible = trackName === 'Object Fusion Tracks'
+
+    const anomalousObjectFusionTrackLayer = new THREE.Group();
+    anomalousObjectFusionTrackLayer.name = `Anomalous ${trackName}`
+    anomalousObjectFusionTrackLayer.visible = false;
+
+		for (let ii = 0, len = trackGeometries.bbox.length; ii < len; ii++) {
```
Comment (bcasselsND) [inline demo/objectFusionTracksLoader.js:272]: Indentation? [I hope you're not using tab characters.]

**Review by bcasselsND — APPROVED** [review]
A couple of nits.

**Pull request merged**